search.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/application"
+	"github.com/almighty/almighty-core/models"
+	"github.com/goadesign/goa"
+)
+
+// SearchController implements the search resource.
+type SearchController struct {
+	*goa.Controller
+	db application.DB
+}
+
+// NewSearchController creates a search controller.
+func NewSearchController(service *goa.Service, db application.DB) *SearchController {
+	if db == nil {
+		panic("db must not be nil")
+	}
+	return &SearchController{Controller: service.NewController("SearchController"), db: db}
+}
+
+// Show runs the show action.
+func (c *SearchController) Show(ctx *app.ShowSearchContext) error {
+	var offset int
+	var limit int
+
+	if ctx.PageOffset == nil {
+		offset = 0
+	} else {
+		offset = int(*ctx.PageOffset)
+	}
+
+	if ctx.PageLimit == nil {
+		limit = 100
+	} else {
+		limit = int(*ctx.PageLimit)
+	}
+	if offset < 0 {
+		return ctx.BadRequest(goa.ErrBadRequest(fmt.Sprintf("offset must be >= 0, but is: %d", offset)))
+	}
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		//return transaction.Do(c.ts, func() error {
+		result, c, err := appl.SearchItems().SearchFullText(ctx.Context, *ctx.Q, &offset, &limit)
+		count := int(c)
+		if err != nil {
+			switch err := err.(type) {
+			case models.BadParameterError:
+				return ctx.BadRequest(goa.ErrBadRequest(fmt.Sprintf("Error listing work items: %s", err.Error())))
+			default:
+				log.Printf("Error listing work items: %s", err.Error())
+				return ctx.InternalServerError()
+			}
+		}
+
+		response := app.SearchResponse{
+			Links: &app.SearchPagingLinks{},
+			Meta:  &app.SearchResponseMeta{TotalCount: float64(count)},
+			Data:  result,
+		}
+
+		// prev link
+		if offset > 0 && count > 0 {
```

Comment:
Hi @baijum , @tsmaeder ,
It would be nice to have it factored out and modularize the code for that same instead of repetitive code. Could we fix it as part of a separate PR , merged to master , and then changes taken into this ( and other PRs ) ?

tracking this here #431